File: Desktop/src/api/site.ts
import { api } from './axios';

export interface SiteSettings {
  platformName: string;
  platformLogoUrl: string | null;
  supportEmail: string | null;
  supportPhone: string | null;
  defaultCurrency: string;
  defaultCountry: string;
  minPasswordLength: number;
  registrationOpen: boolean;
  maintenanceMode: boolean;
}

export interface Country {
  code: string;
  name: string;
  currency: string;
  dialCode: string;
}

export interface PublicPlan {
  code: string;
  name: string;
  description?: string;
  price: {
    amount: number;
    currency: string;
    interval: 'once' | 'month' | 'year';
  };
  limits: {
    maxOwners: number;
    maxManagers: number;
    maxCashiers: number;
    maxProducts: number;
    maxTransactionsPerMonth: number;
    maxAiCallsPerDay: number;
  };
  features: {
    aiInsights: boolean;
    multiLocation: boolean;
    api: boolean;
    prioritySupport: boolean;
    customDomain: boolean;
  };
  trialDays: number;
}

export const siteApi = {
  getSettings: () =>
    api.get<{ data: SiteSettings }>('/public/site/settings').then((r) => r.data.data),

  getBusinessTypes: () =>
    api.get<{ data: string[] }>('/public/site/business-types').then((r) => r.data.data),

  getCountries: () =>
    api.get<{ data: Country[] }>('/public/site/countries').then((r) => r.data.data),

  getCurrencies: () =>
    api.get<{ data: string[] }>('/public/site/currencies').then((r) => r.data.data),

  getLegalLinks: () =>
    api
      .get<{ data: Record<string, string> }>('/public/site/legal-links')
      .then((r) => r.data.data),

  getFeatureFlags: () =>
    api
      .get<{ data: { registrationOpen: boolean; maintenanceMode: boolean } }>(
        '/public/site/feature-flags'
      )
      .then((r) => r.data.data),

  getPlans: () =>
    api.get<{ data: PublicPlan[] }>('/public/site/plans').then((r) => r.data.data),
};

File: Android/client/lib/api/site.dart
import 'api_client.dart';

class SiteApi {
  static final _api = BizOsApiClient.instance;

  static Future<Response<dynamic>> settings() =>
      _api.get('/public/site/settings');
  static Future<Response<dynamic>> businessTypes() =>
      _api.get('/public/site/business-types');
  static Future<Response<dynamic>> countries() =>
      _api.get('/public/site/countries');
  static Future<Response<dynamic>> currencies() =>
      _api.get('/public/site/currencies');
  static Future<Response<dynamic>> legalLinks() =>
      _api.get('/public/site/legal-links');
  static Future<Response<dynamic>> featureFlags() =>
      _api.get('/public/site/feature-flags');
  static Future<Response<dynamic>> plans() => _api.get('/public/site/plans');
}
